opensearch_py_ml/ml_commons/rag_pipeline/rag/AIConnectorHelper.py

```diff
@@ -0,0 +1,698 @@
+            RoleSessionName=role_session_name,
+        )
+
+        # Obtain the temporary credentials from the assumed role 
+        temp_credentials = assumed_role_object["Credentials"]
+
+        return temp_credentials
+    
+    def get_ml_auth(self, create_connector_role_name):
+        """
+        Obtain AWS4Auth credentials for ML API calls using the specified IAM role.
+        """
+        create_connector_role_arn = self.get_role_arn(create_connector_role_name)
+        if not create_connector_role_arn:
+            raise Exception(f"IAM role '{create_connector_role_name}' not found.")
+
+        temp_credentials = self.assume_role(create_connector_role_arn)
+        awsauth = AWS4Auth(
+            temp_credentials["AccessKeyId"],
+            temp_credentials["SecretAccessKey"],
+            self.region,
+            'es',
+            session_token=temp_credentials["SessionToken"],
+        )
+        return awsauth
+
+    def create_connector(self, create_connector_role_name, payload):
+        create_connector_role_arn = self.get_role_arn(create_connector_role_name)
+        temp_credentials = self.assume_role(create_connector_role_arn)
+        awsauth = AWS4Auth(
+            temp_credentials["AccessKeyId"],
+            temp_credentials["SecretAccessKey"],
+            self.region,
+            'es',
+            session_token=temp_credentials["SessionToken"],
+        )
+
+        path = '/_plugins/_ml/connectors/_create'
+        url = self.opensearch_domain_url + path
+
+        headers = {"Content-Type": "application/json"}
+
+        r = requests.post(url, auth=awsauth, json=payload, headers=headers)
+        print(r.text)
+        connector_id = json.loads(r.text)['connector_id']
+        return connector_id
+    
+    def search_model_group(self, model_group_name):
+        payload = {
+            "query": {
+                "term": {
+                    "name.keyword": {
+                        "value": model_group_name
+                    }
+                }
+            }
+        }
+        headers = {"Content-Type": "application/json"}
+        
+        # Obtain temporary credentials
+        create_connector_role_arn = self.get_role_arn('my_test_create_bedrock_connector_role')  # Replace with actual role name
+        temp_credentials = self.assume_role(create_connector_role_arn)
+        awsauth = AWS4Auth(
+            temp_credentials["AccessKeyId"],
+            temp_credentials["SecretAccessKey"],
+            self.region,
+            'es',
+            session_token=temp_credentials["SessionToken"],
+        )
+        
+        r = requests.post(
+            f'{self.opensearch_domain_url}/_plugins/_ml/model_groups/_search',
+            auth=awsauth,
+            json=payload,
+            headers=headers
+        )
+        
+        response = json.loads(r.text)
+        return response
+    
+    def create_model_group(self, model_group_name, description, create_connector_role_name):
```

Comment:
Can we leverage [these methods](https://github.com/opensearch-project/opensearch-py-ml/blob/main/opensearch_py_ml/ml_commons/model_access_control.py)?

Reply:
leveraged!